### PR TITLE
ROX-10832, ROX-10833: Enable Go linters

### PR DESCRIPTION
**Pull request by vladbologa**
Enables the following linters, and adds the formatting changes they made to the code:
- [ROX-10832](https://issues.redhat.com//browse/ROX-10832): enable gofmt and goimports
- [ROX-10833](https://issues.redhat.com//browse/ROX-10833): enable gocritic

## Description

The changes made here are all automatic changes done by the respective linters.

## Checklist (Definition of Done)

- [ ] ~~Unit and integration tests added~~
- [ ] ~~Documentation added if necessary~~
- [x] CI and all relevant tests are passing

## Test manual

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```

### CODE REVIEW

File: fleetshard/pkg/central/reconciler/reconciler_test.go [hunk 1]
```diff
@@ -11,13 +12,14 @@ import (
 	"github.com/stackrox/acs-fleet-manager/internal/dinosaur/pkg/api/private"
 	"github.com/stackrox/rox/operator/apis/platform/v1alpha1"
 
+	"testing"
```
Comment (mtesseract) [inline fleetshard/pkg/central/reconciler/reconciler_test.go:15]: What's the reason for keeping this separate from the above `"context"`? Maybe I don't have a clear picture of the exact style guidelines in my head, but I thought it's a common style to have the imports from the standard library at the top and the other below.

File: fleetshard/pkg/runtime/runtime.go [hunk 2]
```diff
@@ -2,8 +2,10 @@ package runtime
 
 import (
 	"context"
+
 	centralReconciler "github.com/stackrox/acs-fleet-manager/fleetshard/pkg/central/reconciler"
 	"github.com/stackrox/acs-fleet-manager/fleetshard/pkg/k8s"
+
 	"time"
```
Comment (mtesseract) [inline fleetshard/pkg/runtime/runtime.go:9]: Same question as above (why not moving this up?).

File: pkg/shared/get_project_root_dir.go [hunk 3]
```diff
@@ -1,3 +1,4 @@
+//go:build !windows
 // +build !windows
```
Comment (mtesseract) [inline pkg/shared/get_project_root_dir.go:2]: Do we need both directives here?

**Review by mtesseract — COMMENTED** [review]
Looking good, thanks! I have just two or three questions.

File: fleetshard/pkg/central/reconciler/reconciler_test.go (hunk 1 above)
Comment (vladbologa) [reply fleetshard/pkg/central/reconciler/reconciler_test.go]: It doesn't look like goimports enforces this. I changed it like you suggested.

File: fleetshard/pkg/runtime/runtime.go (hunk 2 above)
Comment (vladbologa) [reply fleetshard/pkg/runtime/runtime.go]: Same as above, I moved it up.

**Review by mtesseract — APPROVED** [review]
lgtm!

**Pull request merged**